src/server/scripts/Argus/AntorusTheBurningThrone/boss_garothi_worldbreaker.cpp

```diff
@@ -0,0 +1,884 @@
+                    break;
+                case EVENT_CANNON_CHOOSER:
+                    DoCastSelf(SPELL_CANNON_CHOOSER);
+                    events.Repeat(16s);
+                    break;
+                case EVENT_SURGING_FEL:
+                {
+                    GuidSet guids = _surgingFelDummyGuids;
+                    guids.erase(_lastSurgingFelDummyGuid);
+                    _lastSurgingFelDummyGuid = Trinity::Containers::SelectRandomContainerElement(guids);
+                    if (Creature* dummy = ObjectAccessor::GetCreature(*me, _lastSurgingFelDummyGuid))
+                        dummy->CastSpell(dummy, SPELL_SURGING_FEL_AREA_TRIGGER);
+
+                    events.Repeat(8s);
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+
+        if (me->GetVictim() && me->GetVictim()->IsWithinMeleeRange(me))
+            DoMeleeAttackIfReady();
+        else
+            DoSpellAttackIfReady(SPELL_CARNAGE);
+    }
+ private:
+     uint8 _apocalypseDriveCount;
+     uint32 _searingBarrageSpellId;
+     uint32 _lastCanonEntry;
+     bool _castEradication;
+     ObjectGuid _lastSurgingFelDummyGuid;
+     GuidSet _surgingFelDummyGuids;
+
+     void CleanupEncounter()
+     {
+         if (Creature* decimator = instance->GetCreature(DATA_DECIMATOR))
+             instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, decimator);
+
+         if (Creature* annihilator = instance->GetCreature(DATA_ANNIHILATOR))
+             instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, annihilator);
+
+         instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_DECIMATION_WARNING);
+         instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_FEL_BOMBARDMENT_WARNING);
+         instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_FEL_BOMBARDMENT_PERIODIC);
+         summons.DespawnAll();
+     }
+
+     void HideCannons()
+     {
+         if (Creature* decimator = instance->GetCreature(DATA_DECIMATOR))
+         {
+             instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, decimator);
+             decimator->AddUnitFlag(UnitFlags(UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_UNK_31));
+         }
+
+         if (Creature* annihilator = instance->GetCreature(DATA_ANNIHILATOR))
+         {
+             instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, annihilator);
+             annihilator->AddUnitFlag(UnitFlags(UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_UNK_31));
+         }
+     }
+};
+
+struct at_garothi_annihilation : AreaTriggerAI
+{
+    at_garothi_annihilation(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        _playerCount = 0;
+    }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        if (!unit->IsPlayer())
+            return;
+
+        _playerCount++;
+
+        if (Creature* annihilation = at->FindNearestCreature(NPC_ANNIHILATION, 3.0f, true))
```

Comment:
The annihilation npc is the summoner of the areatrigger right?
Isn't it possible to get the summoner of this areatrigger in that case?

Reply:
The areatrigger AI class does not have a summoner member. That's why I had to go this way.

Reply:
at->GetCaster()